Src/Actions/DisplayMessageAction.lua

```diff
@@ -0,0 +1,42 @@
+local TextService = game:GetService("TextService")
+
+local Action = require(script.Parent.Parent.API.Action)
+
+return function()
+	local DisplayMessageAction = Action.new({
+		Name = "Display Message Action",
+		Uuid = "internal:display-message-action",
+
+		Arguments = {
+			{
+				ArgumentName = "Target Player Id",
+				ArgumentType = "number",
+				ArgumentDefault = -1,
+				ArgumentIsOptional = true,
+			},
+			{
+				ArgumentName = "Server Message",
+				ArgumentType = "string",
+				ArgumentDefault = "Hello, World!",
+				ArgumentIsOptional = false,
+			}
+		}
+	})
+	
+	function DisplayMessageAction:OnRun(targetPlayerId: number?, serverMessage: string)
+		local textFilterResult: TextFilterResult = TextService:FilterStringAsync(serverMessage, 1, Enum.TextFilterContext.PublicChat)
+		local filteredServerMessage
+
+		if targetPlayerId then
+			filteredServerMessage = textFilterResult:GetNonChatStringForUserAsync(targetPlayerId)
```

Comment:
is it possible to not display the message if its just ############# or like 70% ###########

Reply:
That could be problematic as our users would then think it's just not working? Not sure what is meant by 70% though

Reply:
You are probably right. Maybe just display "This message was filtered" if it is only tags. By 70% I meant the majority of the message was filtered so doesn't have any useful information just 'Hello ############# update ##########'